From d0f8c74a7864786cf040da31c7db83d8d2fcc5e4 Mon Sep 17 00:00:00 2001
From: Michael Webster <miketwebster@gmail.com>
Date: Mon, 15 Jan 2024 15:16:08 -0500
Subject: [PATCH] gnome-idle-monitor.c: Fix some dbus names Mutter->Muffin.

The session manager uses this to get notifications about idle
state changes now - instead of its own idle-monitor code (which
only worked with x11), it now relies on muffin to notify on
changes from various sources (xorg, wayland, upower).
---
 libcinnamon-desktop/gnome-idle-monitor.c | 6 +++---
 libcinnamon-desktop/idle-monitor.xml     | 4 ++--
 libcinnamon-desktop/meson.build          | 2 +-
 3 files changed, 6 insertions(+), 6 deletions(-)

diff --git a/libcinnamon-desktop/gnome-idle-monitor.c b/libcinnamon-desktop/gnome-idle-monitor.c
index 71d8918..81aaaab 100644
--- a/libcinnamon-desktop/gnome-idle-monitor.c
+++ b/libcinnamon-desktop/gnome-idle-monitor.c
@@ -67,7 +67,7 @@ G_DEFINE_TYPE_WITH_CODE (GnomeIdleMonitor, gnome_idle_monitor, G_TYPE_OBJECT,
              G_IMPLEMENT_INTERFACE (G_TYPE_INITABLE,
                         gnome_idle_monitor_initable_iface_init))
 
-#define IDLE_MONITOR_PATH "/org/gnome/Mutter/IdleMonitor/Core"
+#define IDLE_MONITOR_PATH "/org/cinnamon/Muffin/IdleMonitor/Core"
 
 static void
 on_watch_fired (MetaDBusIdleMonitor *proxy,
@@ -256,7 +256,7 @@ on_name_appeared (GDBusConnection *connection,
     meta_dbus_object_manager_client_new (connection,
                          G_DBUS_OBJECT_MANAGER_CLIENT_FLAGS_NONE,
                          name_owner,
-                         "/org/gnome/Mutter/IdleMonitor",
+                         "/org/cinnamon/Muffin/IdleMonitor",
                          monitor->priv->cancellable,
                          on_object_manager_ready,
                          monitor);
@@ -296,7 +296,7 @@ gnome_idle_monitor_initable_init (GInitable     *initable,
     monitor = GNOME_IDLE_MONITOR (initable);
 
     monitor->priv->name_watch_id = g_bus_watch_name (G_BUS_TYPE_SESSION,
-                             "org.gnome.Mutter.IdleMonitor",
+                             "org.cinnamon.Muffin.IdleMonitor",
                              G_BUS_NAME_WATCHER_FLAGS_NONE,
                              on_name_appeared,
                              on_name_vanished,
diff --git a/libcinnamon-desktop/idle-monitor.xml b/libcinnamon-desktop/idle-monitor.xml
index 4bd69cb..5203883 100644
--- a/libcinnamon-desktop/idle-monitor.xml
+++ b/libcinnamon-desktop/idle-monitor.xml
@@ -3,14 +3,14 @@
 'http://www.freedesktop.org/standards/dbus/1.0/introspect.dtd'>
 <node>
   <!--
-      org.gnome.Mutter.IdleMonitor:
+      org.cinnamon.Muffin.IdleMonitor:
       @short_description: idle monitor interface
 
       This interface is used by gnome-desktop to implement
       user activity monitoring.
   -->
 
-  <interface name="org.gnome.Muffin.IdleMonitor">
+  <interface name="org.cinnamon.Muffin.IdleMonitor">
     <method name="GetIdletime">
       <arg name="idletime" direction="out" type="t"/>
     </method>
diff --git a/libcinnamon-desktop/meson.build b/libcinnamon-desktop/meson.build
index fbab37a..61976c9 100644
--- a/libcinnamon-desktop/meson.build
+++ b/libcinnamon-desktop/meson.build
@@ -1,7 +1,7 @@
 dbus_idle_built_sources = gnome.gdbus_codegen('meta-dbus-idle-monitor',
   'idle-monitor.xml',
   namespace: 'MetaDBus',
-  interface_prefix: 'org.gnome.Muffin',
+  interface_prefix: 'org.cinnamon.Muffin',
   object_manager: true
 )
 
